fabric-particles-v1/src/main/java/net/fabricmc/fabric/impl/client/particle/FabricParticleManager.java

```diff
@@ -81,19 +90,38 @@ public boolean loadParticle(ResourceManager manager, Identifier id) {
 				throw new IllegalStateException("(Fabric) Missing texture list for particle " + id);
 			}
 
-			provider.setSprites(spriteIds);
+			unloadedSprites.put(id, spriteIds.stream()
+					.map(sprite -> new Identifier(sprite.getNamespace(), "particle/" + sprite.getPath()))
+					.collect(Collectors.toList())
+			);
 		} catch (IOException e) {
-			throw new IllegalStateException("Failed to load description for particle " + id, e);
+			throw new IllegalStateException("(Fabric) Failed to load description for particle " + id, e);
 		}
 
 		return true; // i got dis
 	}
 
+	public CompletableFuture<Void> reload(CompletableFuture<Void> prev, ResourceReloadListener.Synchronizer sync, ResourceManager manager, Profiler executeProf, Profiler applyProf, Executor one, Executor two) {
+		return prev.thenCompose(sync::whenPrepared).thenAcceptAsync(data -> {
```

Comment:
When this is called the previous stage is already like apply finished, so you don't need `thenCompose`, `thenAcceptAsync` should suffice

Reply:
I'm going to apply this change, however I won't be able to test it right now so I'm trusting @liach to know what he's talking about.